homeassistant/components/orvibo/switch.py

```diff
@@ -1,120 +1,170 @@
-
-    for host, data in switch_data.items():
-        try:
-            switches.append(
-                S20Switch(data.get(CONF_NAME), S20(host, mac=data.get(CONF_MAC)))
+    """Set up the integration from configuration.yaml."""
+    for switch in config.get("switches", []):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=switch,
+        )
+
+        if (
+            result.get("type") is FlowResultType.ABORT
+            and result.get("reason") != "already_configured"
+        ):
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                f"yaml_deprecation_import_issue_{switch.get('mac').replace(':', '').lower()}",
+                breaks_in_ha_version="2026.11.0",
+                is_fixable=False,
+                issue_domain=DOMAIN,
+                severity=ir.IssueSeverity.WARNING,
+                translation_key="yaml_deprecation_import_issue",
+                translation_placeholders={
+                    "host": switch.get("host"),
+                    "mac": switch.get("mac"),
+                    "name": switch.get("name"),
+                },
             )
-            _LOGGER.debug("Initialized S20 at %s", host)
-        except S20Exception:
-            _LOGGER.error("S20 at %s couldn't be initialized", host)
-
-    add_entities_callback(switches)
+            return
+
+        ir.async_create_issue(
+            hass,
+            DOMAIN,
+            f"eyaml_deprecation_{switch.get('mac').replace(':', '').lower()}",
+            breaks_in_ha_version="2026.11.0",
+            is_fixable=False,
+            is_persistent=True,
+            severity=ir.IssueSeverity.WARNING,
+            translation_key="yaml_deprecation",
+            translation_placeholders={
+                "host": switch.get("host"),
+                "mac": switch.get("mac"),
+                "name": switch.get("name"),
+            },
+        )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: S20ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Setup Entry."""
+    switch = []
+    switch.append(
+        S20Switch(entry.data[CONF_NAME], entry.data[CONF_HOST], entry.data[CONF_MAC]),
+    )
+    async_add_entities(switch)
 
 
 class S20Switch(SwitchEntity):
     """Representation of an S20 switch."""
 
-    def __init__(self, name, s20):
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = True
+
+    def __init__(self, name, host, mac):
         """Initialize the S20 device."""
 
         self._name = name
-        self._s20 = s20
+        self._host = host
+        self._mac = mac
         self._state = False
         self._exc = S20Exception
-
-    @property
-    def name(self):
-        """Return the name of the switch."""
-        return self._name
+        self._s20 = S20(self._host, self._mac)
```

Comment:
1. With moving to config entry, we want to move the creation and the check if a device is connectable to `__init__.py`. If we can't connect, we should raise `ConfigEntryNotReady` to retry later.
2. So previously we excepted the `S20Exception` when creating the `S20` object, so I think that the object itself does blocking IO. Which means we should run this in an executor job